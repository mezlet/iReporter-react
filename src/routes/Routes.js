import React from "react";
import { Route, Switch } from "react-router-dom";
import { createBrowserHistory } from "history";
import HomePage from "../components/containers/HomePage/Home";
import connectSignup from "../components/containers/SignUp/SignUp";
import connectLogin from "../components/containers/Login/Login";
import CreateReport from "../components/containers/CreateReport/CreateReport";
import Profile from "../components/containers/Profile/Profile";
import Record from "../components/containers/Record/Record";
import NewReportView from "../components/containers/NewReportView/NewReportView";
import EditReport from "../components/containers/EditReport/EditReport";

const history = createBrowserHistory();

const route = () => (
  <Route history={history}>
    <Switch>
      <Route path="/" component={HomePage} exact />
      <Route path="/signup" component={connectSignup} exact />
      <Route path="/login" component={connectLogin} exact />
      <Route path="/create" component={CreateReport} exact />
      <Route path="/profile" component={Profile} exact />
      <Route path="/record" component={Record} exact />
      <Route path="/edit-incident/:id" component={EditReport} exact />
      <Route path="/new-report-view/:id" component={NewReportView} exact />
    </Switch>
  </Route>
);

export default route;
