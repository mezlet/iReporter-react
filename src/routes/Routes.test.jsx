import React from 'react';
import { shallow } from 'enzyme';
import Routes from './Routes';

const wrapper = shallow(<Routes />);

describe('<Routes/> rendering', () => {
  it('should render without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
