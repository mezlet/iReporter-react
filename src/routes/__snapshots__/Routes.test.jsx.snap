// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Routes/> rendering should render without crashing 1`] = `ShallowWrapper {}`;
